server/controllers/postOrder.js

```diff
@@ -0,0 +1,35 @@
+const { insertOrder, insertOrderMeals } = require('../database/queries/order');
+const orderSchema = require('../validation/orderSchema');
+
+module.exports = (req, res, next) => {
+  let allMeals;
+  orderSchema
+    .validateAsync({
+      ...req.body, tableNumber: req.user.tableNumber,
+    })
+    .then((data) => {
+      const {
+        createdAt, totalPrice, meals,
+      } = data;
+      allMeals = meals;
+      return insertOrder(createdAt, totalPrice, req.user.tableNumber);
+    })
+    .then((order) => {
+      const { rows } = order;
+      const orderId = rows[0].id;
+      return Promise.all(allMeals.map((meal) => insertOrderMeals(orderId, meal)));
+    }).then(() => {
+      res.status(201).send({ statusCode: 201, message: 'Insertion order success' });
+    })
+    .catch((error) => {
+      if (error.code === '23503') {
+        res.status(422).send({ statusCode: 422, error: error.detail });
```

Comment:
Don't send the error generated by the database!!
Write your own error message, say something like this:
```suggestion
        res.status(422).send({ statusCode: 422, error: Unable to process your order. Meal does not exist'' });
```